src/mozillavpn.cpp

```diff
@@ -104,11 +104,14 @@ MozillaVPN::MozillaVPN() : m_private(new Private()) {
   connect(&m_alertTimer, &QTimer::timeout, [this]() { setAlert(NoAlert); });
 
   connect(&m_periodicOperationsTimer, &QTimer::timeout, [this]() {
-    scheduleTask(new TaskAccountAndServers());
-    scheduleTask(new TaskCaptivePortalLookup());
-    scheduleTask(new TaskHeartbeat());
-    scheduleTask(new TaskSurveyData());
-    scheduleTask(new TaskGetFeatureList());
+    if (controller()->state() != Controller::State::StateConnecting &&
```

Comment:
In theory this should not be needed. The tasks can be schedule at any time. They do not run during these states because we should have a TaskControllerAction to prevent it.